woocommerce/google-listings-and-ads
# Externalize all `@woocommerce` packages via DEWP and remove the selective bundling implementation

### Changes proposed in this Pull Request:

Closes #1527
Closes #1971

This PR externalizes all `@woocommerce` packages via DEWP, and it can also fix #1971 by the way.

- Change the bundled `@woocommerce` packages to use external sources from DEWP.
   - `@woocommerce/currency`
   - `@woocommerce/date`
   - `@woocommerce/number`
   - `@woocommerce/tracks`
- Remove the selective bundling implementation that gradually externalizes packages into DEWP.
- Rebalance the max size config of bundlewatch for each non-wildcard file to 1.2 times.
- Update DEWP document.

### Screenshots:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/baf0110b-3aff-4a96-b344-ea1157b04d2d)

### Detailed test instructions:

1. Check if all build scripts, `npm start`, `npm run dev`, `npm run start:hot` and `npm run build`, can run successfully.
2. Run bundle analyzer: `npm run env -- NODE_ENV=production wp-scripts build --webpack-bundle-analyzer`
   - Check if there is no DEWPable `@woocommerce/*` package included in the build bundle.
3. Read through [Selective bundling & extracting](https://github.com/woocommerce/google-listings-and-ads/blob/6358de365713ede2013f4edae6d2d680ead09f3e/Working%20with%20DEWP.md#selective-bundling--extracting) to see if the updated description is appropriate.
4. Follow the reproduction steps in #1971 to check if `wp.date.getDate()` can run successfully.

### Additional details:

The [current versions](https://github.com/woocommerce/google-listings-and-ads/blob/6358de365713ede2013f4edae6d2d680ead09f3e/package.json#L22-L28) of `@woocommerce/currency`, `@woocommerce/date`, `@woocommerce/number` and `@woocommerce/tracks` packages are the same versions as the latest WooCoomerce release (8.0.2):
- https://github.com/woocommerce/woocommerce/blob/8.0.2/packages/js/currency/CHANGELOG.md
- https://github.com/woocommerce/woocommerce/blob/8.0.2/packages/js/date/CHANGELOG.md
- https://github.com/woocommerce/woocommerce/blob/8.0.2/packages/js/number/CHANGELOG.md
- https://github.com/woocommerce/woocommerce/blob/8.0.2/packages/js/tracks/CHANGELOG.md

Therefore, switching to DEWP should not have any effect.

### Changelog entry

> Dev - Externalize all WooCommerce JavaScript packages via Dependency Extraction Webpack Plugin (DEWP) and remove the selective bundling implementation that gradually externalizes packages into DEWP.
